slu/slu/utils/config.py

```diff
@@ -172,6 +174,132 @@ def generate(self) -> Dict[str, Config]:
         return {config_dict[const.MODEL_NAME]: config}
 
 
+class YAMLPromptConfig(ConfigDataProviderInterface):
+    """
+    An instance of this class will load, validate and fetch the state <-> prompts mapping from prompts.yaml
+    The instance will be further passed to classifier plugins like xlmr.py.
+    """
+
+    def __init__(
+        self, config_path: Optional[str] = None, debug: Optional[bool] = True
+    ) -> None:
+        self.config_path: str = config_path or const.PROMPTS_CONFIG_PATH
+        self.null_prompt_token: str = "<pad>"
+        self.missing_nls_labels: set = set()
+        self.config_dict: dict[str] = {}
+        self.supported_languages: list[str] = []
+        self.debug: bool = debug
+
+    def get_config_dict(self, config_path: str) -> dict:
+        error_message = "Unable to read prompts.yaml file, ensure correct format."
```

Comment:
[Optional] If you are not reusing this error message, there is no need for a separate variable, you can directly put it in the `TypeError`.

Reply:
Removed the variable, wasn't reusing it anywhere else